core/services/ocr2/plugins/directrequestocr/aggregation.go

```diff
@@ -0,0 +1,102 @@
+package directrequestocr
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+	"sort"
+
+	"github.com/smartcontractkit/chainlink/core/services/ocr2/plugins/directrequestocr/config"
+)
+
+func CanAggregate(N int, F int, observations []*ProcessedRequest) bool {
+	return len(observations) >= N-F
+}
+
+func Aggregate(aggMethod config.AggregationMethod, observations []*ProcessedRequest) (*ProcessedRequest, error) {
+	if len(observations) == 0 {
+		return nil, fmt.Errorf("empty observation list passed for aggregation")
+	}
+	var errored []*ProcessedRequest
+	var successful []*ProcessedRequest
+	finalResult := ProcessedRequest{
+		RequestID: observations[0].RequestID,
+		Result:    []byte{},
+		Error:     []byte{},
+	}
+	for _, obs := range observations {
```

Comment:
Shall we do a sanity check that `observations[index].RequestID` are equal for all observations?

Reply:
Sure, why not.